### PR TITLE
pcapplusplus: add version v22.11

### PR DESCRIPTION
**Pull request by seladb**
Specify library name and version:  **pcapplusplus/22.11**
- Update PcapPlusPlus to v22.11

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by seladb** [comment]
@prince-chrismc this is the PR with the fix we discussed here: https://github.com/conan-io/conan-center-index/pull/14159

**Comment by seladb** [comment]
> ### Conan v1 pipeline
> 
> Failure in build 4 (`ddca22d9f845eb903ab7a88efd889a5b770d1191`):
> 
>  * **pcapplusplus/21.05@:**
>    Visual Studio license has expired, infrastructure team has been notified and will take care of it.
> 
>  * **pcapplusplus/22.05@:**
>    Didn't run or was cancelled before finishing
> 
>  * **pcapplusplus/21.11@:**
>    Didn't run or was cancelled before finishing
> 
>  * **pcapplusplus/22.11@:**
>    Didn't run or was cancelled before finishing
> 
> ---
> 
> <sup>**Note:** To save resources, CI tries to finish as soon as an error is found. For this reason you might find that not all the references have been launched or not all the configurations for a given reference. Also, take into account that we cannot guarantee the order of execution as it depends on CI workload and workers availability.</sup>

Conan team - can you help with this error? Maybe we should rerun it?

**Comment by seladb** [comment]
Thank you @prince-chrismc ! I think the PR is ready to review

File: recipes/pcapplusplus/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/pcapplusplus/all/test_v1_package/CMakeLists.txt:8]:
```suggestion
include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
CONAN_LIBS is no longer used. Please, use CMake targets always.

File: recipes/pcapplusplus/all/test_v1_package/test_package.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#include <iostream>
```
Comment (uilianries) [inline recipes/pcapplusplus/all/test_v1_package/test_package.cpp:1]: This file can be removed. Re-use directly from test_package folder.

**Review by uilianries — COMMENTED** [review]
Please consider https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates/cmake_package/all as template for test_package and test_v1_package.

File: recipes/pcapplusplus/all/test_v1_package/test_package.cpp (hunk 2 above)
Comment (seladb) [reply recipes/pcapplusplus/all/test_v1_package/test_package.cpp]: Done in https://github.com/conan-io/conan-center-index/pull/14712/commits/4a0959d18aaeb0d995400b3cff6ccea91b38b080

**Comment by seladb** [comment]
> ### Conan v1 pipeline
> 
> Failure in build 11 (`1de5980b938e5558100441f947a4863be17986f1`):
> 
>     * **pcapplusplus/21.05@:**
>       Didn't run or was cancelled before finishing
> 
>     * **pcapplusplus/21.11@:**
>       Didn't run or was cancelled before finishing
> 
>     * **pcapplusplus/22.05@:**
>       CI failed to create some packages ([All logs](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/14712/11-linux-gcc/pcapplusplus/22.05//summary.json))
>       Logs for packageID `0ad9b72ff7ca8d391072b0c8f15c8cfb5fa0dd3c`:
>       ```
>       [settings]
>       arch=x86_64
>       build_type=Release
>       compiler=gcc
>       compiler.libcxx=libstdc++11
>       compiler.version=7
>       os=Linux
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       ```
>       [...]
>       -- Library Pcap++ found /home/conan/w/prod/BuildSingleReference/.conan/data/pcapplusplus/22.05/_/_/package/0ad9b72ff7ca8d391072b0c8f15c8cfb5fa0dd3c/lib/libPcap++.a
>       -- Library Packet++ found /home/conan/w/prod/BuildSingleReference/.conan/data/pcapplusplus/22.05/_/_/package/0ad9b72ff7ca8d391072b0c8f15c8cfb5fa0dd3c/lib/libPacket++.a
>       -- Library Common++ found /home/conan/w/prod/BuildSingleReference/.conan/data/pcapplusplus/22.05/_/_/package/0ad9b72ff7ca8d391072b0c8f15c8cfb5fa0dd3c/lib/libCommon++.a
>       -- Library pcap found /home/conan/w/prod/BuildSingleReference/.conan/data/libpcap/1.9.1/_/_/package/d50a0d523d98c15bb147b18fa7d203887c38be8b/lib/libpcap.a
>       -- Conan: Adjusting default RPATHs Conan policies
>       -- Conan: Adjusting language standard
>       -- Conan: Compiler GCC>=5, checking major version 7
>       -- Conan: Checking correct version: 7
>       -- Conan: C++ stdlib: libstdc++11
>       -- Conan: called by CMake conan helper
>       -- Conan: Adjusting output directories
>       -- Conan: Using cmake global configuration
>       -- Conan: Adjusting default RPATHs Conan policies
>       -- Conan: Adjusting language standard
>       -- Conan: Compiler GCC>=5, checking major version 7
>       -- Conan: Checking correct version: 7
>       -- Conan: C++ stdlib: libstdc++11
>       -- Configuring done
>       -- Generating done
>       -- Build files have been written to: /home/conan/w/prod/BuildSingleReference/conan-center-index/recipes/pcapplusplus/all/test_v1_package/build/ac8e5b135bec2d2511377cc6fdda763812fc2867
>       
>       ----Running------
>       > cmake --build '/home/conan/w/prod/BuildSingleReference/conan-center-index/recipes/pcapplusplus/all/test_v1_package/build/ac8e5b135bec2d2511377cc6fdda763812fc2867' '--' '-j3'
>       -----------------
>       Scanning dependencies of target test_package
>       [ 50%] Building CXX object test_package/CMakeFiles/test_package.dir/test_package.cpp.o
>       [100%] Linking CXX executable bin/test_package
>       [100%] Built target test_package
>       pcapplusplus/22.05 (test package): Running test()
>       
>       ----Running------
>       > bin/test_package /home/conan/w/prod/BuildSingleReference/conan-center-index/recipes/pcapplusplus/all/test_v1_package/1_packet.pcap
>       -----------------
>       CMake Warning:
>         Manually-specified variables were not used by the project:
>       
>           CMAKE_EXPORT_NO_PACKAGE_REGISTRY
>           CMAKE_INSTALL_BINDIR
>           CMAKE_INSTALL_DATAROOTDIR
>           CMAKE_INSTALL_INCLUDEDIR
>           CMAKE_INSTALL_LIBDIR
>           CMAKE_INSTALL_LIBEXECDIR
>           CMAKE_INSTALL_OLDINCLUDEDIR
>           CMAKE_INSTALL_SBINDIR
>       
>       
>       /bin/sh: 1: bin/test_package: not found
>       ERROR: pcapplusplus/22.05 (test package): Error in test() method, line 38
>       	self.run(f"{bin_path} {pcap_file_path}", run_environment=True)
>       	ConanException: Error 127 while executing bin/test_package /home/conan/w/prod/BuildSingleReference/conan-center-index/recipes/pcapplusplus/all/test_v1_package/1_packet.pcap
>       ```
> 
>     * **pcapplusplus/22.11@:**
>       Didn't run or was cancelled before finishing
> 
> 
> **Note:** To save resources, CI tries to finish as soon as an error is found. For this reason you might find that not all the references have been launched or not all the configurations for a given reference. Also, take into account that we cannot guarantee the order of execution as it depends on CI workload and workers availability.

@uilianries can you help me understand why the test fails? I tried to follow your comments...

**Comment by seladb** [comment]
@uilianries @prince-chrismc can you please guide me on why the build is failing? I followed @uilianries PR comments and it started failing...

**Comment by seladb** [comment]
@uilianries should I revert to the previous commit that works? Please let me know

File: recipes/pcapplusplus/all/test_package/conanfile.py [hunk 3]
```diff
@@ -33,6 +33,7 @@ def test(self):
                     os.path.join(self.deps_cpp_info['libpcap'].bin_paths[0], "pcap.dll"),
                     os.path.join("bin", "wpcap.dll")
                 )
-            bin_path = os.path.join("bin", "test_package")
+
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             pcap_file_path = os.path.join(self.source_folder, "1_packet.pcap")
-            self.run(f"{bin_path} {pcap_file_path}", run_environment=True)
+            self.run(f"{bin_path} {pcap_file_path}", env="conanrun")
```
Comment (prince-chrismc) [inline recipes/pcapplusplus/all/test_package/conanfile.py:39]: I think there might have been some confusion, this has not been migrated so the "old" way should still be used  AFAIK

I would try with as it previously was using

```suggestion
            bin_path = os.path.join("bin", "test_package")
            pcap_file_path = os.path.join(self.source_folder, "1_packet.pcap")
            self.run(f"{bin_path} {pcap_file_path}", run_environment=True)
```

**Comment by seladb** [comment]
@prince-chrismc @uilianries I'm not sure why it doesn't work, can you please assist?
I'm pretty lost here...

**Comment by seladb** [comment]
Thank you @uilianries for all of your help!

Is the PR ready to merge now?

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**